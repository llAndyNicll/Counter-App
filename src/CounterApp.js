import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {


    const [ counter, setCounter ] = useState( value );


    // handleAdd

    const handleAdd = () => {

        setCounter( counter + 1 );


        // Por si no tienes acceso a la primera variante

        // setCounter( (c) => { c + 1 } );

    };

    const handleReset = () => {

        setCounter( value );

    };

    const handleSustract = () => {

        setCounter( counter - 1 );

    };



    return (
        <div>

            <h1>CounterApp</h1>
            <h2>{ counter }</h2>

            <button onClick={ handleAdd } >+1</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSustract } >-1</button>

        </div>
    );

};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};

export default CounterApp;