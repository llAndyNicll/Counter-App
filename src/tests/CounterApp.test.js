import React from "react";
import { shallow } from 'enzyme';
import CounterApp from "../CounterApp";

describe( 'Pruebas del archivo CounterApp.js' , () => {

    let wrapper = shallow( <CounterApp value = { 10 } /> );

    beforeEach( () => {

        wrapper = shallow( <CounterApp value = { 10 } /> );

    });

    test( 'Debe de mostrar <CounterApp /> correctamente.' , () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'Debe de mostrar el valor por defecto de 100.' , () => {

        const value = 100;

        const wrapper = shallow( <CounterApp value = { value } /> );

        const contador = wrapper.find( 'h2' ).text().trim(); // trim elimina espacios

        expect( contador ).toBe( '100' );

    });

    test( 'Debe de incrementar el contador con el boton +1.' , () => {

        const btn1 = wrapper.find('button').at( 0 ).simulate( 'click' );

        // console.log( btn1.html() );

        const contador = wrapper.find( 'h2' ).text().trim();

        // console.log( contador );

        expect( contador ).toBe( '11' );

    });

    test( 'Debe de decrementar el contador con el boton -1.' , () => {

        const btn1 = wrapper.find('button').at( 2 ).simulate( 'click' );

        // console.log( btn1.html() );

        const contador = wrapper.find( 'h2' ).text().trim();

        // console.log( contador );

        expect( contador ).toBe( '9' );

    });

    test( 'Debe de resetiar el contador con el boton Reset.' , () => {;

        const value = 105;

        const wrapper = shallow( <CounterApp value = { value } /> );

        const btn1 = wrapper.find('button').at( 0 ).simulate( 'click' );

        const btn2 = wrapper.find('button').at( 1 ).simulate( 'click' );

        // console.log( btn1.html() );

        const contador = wrapper.find( 'h2' ).text().trim();

        // console.log( contador );

        expect( contador ).toBe( '105' );

    });

});