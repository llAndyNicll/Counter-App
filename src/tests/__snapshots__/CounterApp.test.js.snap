// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del archivo CounterApp.js Debe de mostrar <CounterApp /> correctamente. 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
